packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx

```diff
@@ -26,37 +26,273 @@ import {
+                  config.managementUrl,
+                  serviceState.service.pattern,
+                ),
+              );
+            }}
+          >
+            Launch Service
+          </MenuContentItem>
+        )}
+        {serviceState.registrationResult instanceof ServiceRegistrationFail && (
+          <MenuContentItem
+            onClick={() =>
+              globalServiceRegistrationState.setFailingView(
+                serviceState.registrationResult as ServiceRegistrationFail,
+              )
+            }
+          >
+            View Error
+          </MenuContentItem>
+        )}
+      </MenuContent>
+    );
+  }),
+);
+
+const ServiceFailViewer = observer(
+  (props: {
+    globalBulkServiceRegistrationState: GlobalBulkServiceRegistrationState;
+    failure: ServiceRegistrationResult | undefined;
+  }) => {
+    const { globalBulkServiceRegistrationState, failure } = props;
+    const closeLogViewer = (): void =>
+      globalBulkServiceRegistrationState.setFailingView(undefined);
+    return (
+      <Dialog
+        open={Boolean(failure)}
+        onClose={closeLogViewer}
+        classes={{
+          root: 'editor-modal__root-container',
+          container: 'editor-modal__container',
+          paper: 'editor-modal__content',
+        }}
+      >
+        <Modal darkMode={true} className="editor-modal">
+          <ModalHeader title={guaranteeNonNullable(failure?.service).path} />
+          <ModalBody>
+            {failure instanceof ServiceRegistrationFail && (
+              <TextInputEditor
+                inputValue={failure.errorMessage}
+                isReadOnly={true}
+                language={EDITOR_LANGUAGE.TEXT}
+              />
+            )}
+          </ModalBody>
+          <ModalFooter>
+            <ModalFooterButton text="Close" onClick={closeLogViewer} />
+          </ModalFooter>
+        </Modal>
+      </Dialog>
+    );
+  },
+);
 
 export const RegisterService = observer(
-  (props: { bulkServiceRegistrationState: BulkServiceRegistrationState }) => {
+  (props: {
+    globalBulkServiceRegistrationState: GlobalBulkServiceRegistrationState;
+  }) => {
     const editorStore = useEditorStore();
     const services = editorStore.graphManagerState.graph.ownServices;
-    const [showRegistrationModel, setShowRegistrationModel] = useState(false);
+    const toggleSelectAllServices = (): void => {
+      editorStore.globalBulkServiceRegistrationState.selectAllServices
+        ? editorStore.globalBulkServiceRegistrationState.toggleSelectAllServices(
+            false,
+          )
+        : editorStore.globalBulkServiceRegistrationState.toggleSelectAllServices(
+            true,
+          );
+
+      editorStore.globalBulkServiceRegistrationState.setSelectAll(
+        editorStore.globalBulkServiceRegistrationState.selectAllServices,
+      );
+      editorStore.globalBulkServiceRegistrationState.setNumOfServicesSelected();
+    };
+
+    const selectService = (
```

Comment:
Only thing we need to do here is toggle everything else in terms of calcualtion should come for free